app/controllers/course_management/available_course_groups_controller.rb

```diff
@@ -6,6 +6,7 @@ class AvailableCourseGroupsController < ApplicationController
     before_action :set_lecturers
     before_action :set_available_course_group, only: %i[edit update destroy]
     before_action :authorized?
+    before_action :event_active?, except: %i[edit update]
```

Comment:
Bu event check olayınıda tekilleştirmemiz lazım. Her yerde kullanılıyor ve biz sürekli yazmak zorunda kalıyoruz. Bu PR içinde bunu düzeltmeyelim ama bu konuda bir teknik borç(dept) olacak şekilde bir iş kaydı açar mısın?.

Reply:
Bu event kontrolünün iki boyutu var;
1. authorization yani yetki tarafı, bu PR'daki `available_course` için yapılan bazı işlemler bu dediğimi örneklemek için biçilmiş kaftan, çünkü ekleme çıkarma zamanında değilsek bazı işlemler yapılamıyor. silme işlemi gibi.

2. boyut ise validasyon tarafı, bir taraftan yetki gibi aslında, arayüz tarafında bazı kısıtlamalar yapıyoruz ve izin yoksa veri ekleyemiyor fakat model bu kısıtlamalardan etkilenmiyor bir şekilde arayüz tarafından hack edilse bile model veriyi kayıt etmemeli. 
Bu durumu farklı bir açıdan daha bakarsak `rails c` 'a girip at koşturabiliyoruz istemesekte bazen veri düzeltmelerini console üzerinden yapıyoruz. Bu durumlarda model validasyonu eklemezsek gözden kaçan hatalar yapabiliriz. Bu bakımdan bizimler içinde koruma önlemi olmuş olur. (Gönül isterki `rails c` ile consola girilip hiç birşey yapılmasın ama hayaller ve hayatlar misali bir şekilde girmek gerekiyor.)

2. boyut şimdilik çok elzem değil fakat herşey için değilse bile bazı event'lar için model tarafındada validasyon kontrolü eklemek gerekli.